### PR TITLE
Mimir 2.4 release notes

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
Mimir 2.4 will be cut in 48h. In this PR I'm playing a bit ahead of time, and drafting the release notes (I don't expect major PRs to be merged in the next 2 days).

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/release-notes/v2.4.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  The new configuration option allows you to set different limits between the received query maximum length (`-query-frontend.max-total-query-length`) and the maximum length of partial queries after splitting and sharding (`-store.max-query-length`).
+
+### Helm chart improvements
+
+The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.4 release, we’re also releasing version 3.2 of the Mimir Helm chart.
+
+Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
+
+- Added support for [topologySpreadContraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/).
+- Replaced the default anti affinity rules with topologySpreadContraints for all components which puts less restrictions on where kubernetes can run pods.
```
Comment (dimitarvdimitrov) [inline docs/sources/release-notes/v2.4.md:45]:
```suggestion
- Replaced the default anti-affinity rules with topologySpreadContraints for all components which puts less restrictions on where Kubernetes can run pods.
```

File: docs/sources/release-notes/v2.4.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            topologyKey: "kubernetes.io/hostname"
+          - labelSelector:
+              matchExpressions:
+                - key: app.kubernetes.io/component
+                  operator: In
+                  values:
+                    - store-gateway
+            topologyKey: "kubernetes.io/hostname"
+  ```
+- Updated the anti affinity rules in the sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml) to account for the relaxed default rules.
```
Comment (dimitarvdimitrov) [inline docs/sources/release-notes/v2.4.md:68]: Can you be more specific? 

```suggestion
- Updated the anti affinity rules in the sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml). The sizing plans now enforce that no two pods of the ingester, store-gateway, or alertmanager StatefulSets are scheduled on the same Node. Pods from different StaatefulSets can share a Node.
```

File: docs/sources/release-notes/v2.4.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+- Support for Openshift Route definition for nginx has been added.
```
Comment (dimitarvdimitrov) [inline docs/sources/release-notes/v2.4.md:69]:
```suggestion
- Support for Openshift Route resource for nginx has been added.
```

File: docs/sources/release-notes/v2.4.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+### Helm chart improvements
+
+The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.4 release, we’re also releasing version 3.2 of the Mimir Helm chart.
+
+Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
+
+- Added support for [topologySpreadContraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/).
+- Replaced the default anti affinity rules with topologySpreadContraints for all components which puts less restrictions on where kubernetes can run pods.
+- **Important**: if you are not using the sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml) in production, you must reintroduce pod affinity rules for the store-gateway.
+  Merge the following to your custom values file:
```
Comment (dimitarvdimitrov) [inline docs/sources/release-notes/v2.4.md:47]:
```suggestion
  Merge the following with your custom values file:
```

File: docs/sources/release-notes/v2.4.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  These new configuration options can be used to override the default TLS settings and meet your security policies. Refer to [Securing Grafana Mimir communications with TLS]({{< relref "../operators-guide/secure/securing-communications-with-tls.md" >}}) for more information.
+
+- **Maximum range query length limit**:
+  Mimir 2.4 introduces the new configuration option `-query-frontend.max-total-query-length` to limit the maximum range query length, computed as the query's `end` minus `start` timestamp. This limit is enforced in the query-frontend and defaults to `-store.max-query-length` if unset.
+
+  The new configuration option allows you to set different limits between the received query maximum length (`-query-frontend.max-total-query-length`) and the maximum length of partial queries after splitting and sharding (`-store.max-query-length`).
+
+### Helm chart improvements
+
+The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.4 release, we’re also releasing version 3.2 of the Mimir Helm chart.
```
Comment (dimitarvdimitrov) [inline docs/sources/release-notes/v2.4.md:40]:
```suggestion
The `mimir-distributed` Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.4 release, we’re also releasing version 3.2 of the `mimir-distributed` Helm chart.
```

not sure what we've used in other release notes, but that's how we refer to it in docs. It's less ambiguous which chart this means.

**Review by dimitarvdimitrov — APPROVED** [review]
helm notes LGTM, thanks for writing @krajorama

File: docs/sources/release-notes/v2.4.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+The highlights that follow include the top features, enhancements, and bugfixes in this release. For the complete list of changes, see the [changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
+
+**Note**: If you are upgrading from Grafana Mimir 2.3, review the [list of important changes](#important-changes) that follow.
+
+## Features and enhancements
+
+- **Query-scheduler ring-based service discovery**:
+  The [query-scheduler]({{< relref "../operators-guide/architecture/components/query-scheduler/index.md" >}}) is an optional, stateless component that retains a queue of queries to execute, and distributes the workload to available queriers. The use the query-scheduler, query-frontends and queriers are required to discover the addresses of the query-scheduler instances.
+
+  Other than the DNS-based service discovery, Mimir 2.4 introduces the ring-based service discovery for the query-scheduler. When enabled, the query-schedulers join their own hash ring (similar to other Mimir components), and the query-frontends and queriers discover query-scheduler instances via the ring.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:21]:
```suggestion
  In addition to DNS-based service discovery, Mimir 2.4 introduces the ring-based service discovery for the query-scheduler. When enabled, the query-schedulers join their own hash ring (similar to other Mimir components), and the query-frontends and queriers discover query-scheduler instances via the ring.
```
Do you mean to say, "As an alternative to"? I kind of think so, but you decide. :)

File: docs/sources/release-notes/v2.4.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  The ring-based service discovery makes easier to setup the query-scheduler in environments where you can't easily define a DNS entry resolving to the running query-scheduler instances. Refer to [query-scheduler configuration]({{< relref "../operators-guide/architecture/components/query-scheduler/index.md#configuration" >}}) for more information.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:23]:
```suggestion
  Ring-based service discovery makes it easier to set up the query-scheduler in environments where you can't easily define a DNS entry that resolves to the running query-scheduler instances. For more information, refer to [query-scheduler configuration]({{< relref "../operators-guide/architecture/components/query-scheduler/index.md#configuration" >}}).
```

File: docs/sources/release-notes/v2.4.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+- **New API endpoint exposing per-tenant limits**:
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:25]:
```suggestion
- **New API endpoint exposes per-tenant limits**:
```

File: docs/sources/release-notes/v2.4.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  Mimir 2.4 introduces a new API endpoint, available on all Mimir components that load the runtime configuration, exposing the limits for the authenticated tenant. This new API endpoint can be used when developing custom integrations with Mimir that require looking up the actual limits applied on a given tenant. Refer to [Get tenant limits]({{< relref "../operators-guide/reference-http-api/index.md#get-tenant-limits" >}}) for more information.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:26]:
```suggestion
  Mimir 2.4 introduces a new API endpoint that is available on all Mimir components that load the runtime configuration, and that exposes the limits for the authenticated tenant. You can use this new API endpoint when developing custom integrations with Mimir that require looking up the actual limits that are applied on a given tenant. For more information, refer to [Get tenant limits]({{< relref "../operators-guide/reference-http-api/index.md#get-tenant-limits" >}}).
```

File: docs/sources/release-notes/v2.4.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+- **New TLS configuration options**:
+  Mimir 2.4 introduces new options to configure the accepted TLS cipher suites and minimum version for the HTTP and gRPC clients used between Mimir components, or by Mimir to communicate to external services (e.g. Consul or etcd).
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:29]:
```suggestion
  Mimir 2.4 introduces new options to configure the accepted TLS cipher suites, and the minimum versions for the HTTP and gRPC clients that are used between Mimir components, or by Mimir to communicate to external services such as Consul or etcd.
```

File: docs/sources/release-notes/v2.4.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+  These new configuration options can be used to override the default TLS settings and meet your security policies. Refer to [Securing Grafana Mimir communications with TLS]({{< relref "../operators-guide/secure/securing-communications-with-tls.md" >}}) for more information.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:31]:
```suggestion
  You can use these new configuration options to override the default TLS settings and meet your security policy requirements. For more information, refer to [Securing Grafana Mimir communications with TLS]({{< relref "../operators-guide/secure/securing-communications-with-tls.md" >}}).
```
Do you mean "... and meet your security ..." or "... to meet your security ..."?

File: docs/sources/release-notes/v2.4.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+- **Maximum range query length limit**:
+  Mimir 2.4 introduces the new configuration option `-query-frontend.max-total-query-length` to limit the maximum range query length, computed as the query's `end` minus `start` timestamp. This limit is enforced in the query-frontend and defaults to `-store.max-query-length` if unset.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:34]:
```suggestion
  Mimir 2.4 introduces the new configuration option `-query-frontend.max-total-query-length` to limit the maximum range query length, which is computed as the query's `end` minus `start` timestamp. This limit is enforced in the query-frontend and defaults to `-store.max-query-length` if unset.
```

File: docs/sources/release-notes/v2.4.md (hunk 1 above)
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md]:
```suggestion
- Replaced the default anti-affinity rules with topology spread contraints for all components, which puts fewer restrictions on where kubernetes can run pods.
```

File: docs/sources/release-notes/v2.4.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+### Helm chart improvements
+
+The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.4 release, we’re also releasing version 3.2 of the Mimir Helm chart.
+
+Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
+
+- Added support for [topologySpreadContraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/).
+- Replaced the default anti affinity rules with topologySpreadContraints for all components which puts less restrictions on where kubernetes can run pods.
+- **Important**: if you are not using the sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml) in production, you must reintroduce pod affinity rules for the store-gateway.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md]:
```suggestion
- **Important**: if you are not using the sizing plans (`small.yaml`, `large.yaml`, `capped-small.yaml`, `capped-large.yaml`) in production, you must reintroduce pod affinity rules for the store-gateway.
```

File: docs/sources/release-notes/v2.4.md (hunk 4 above)
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md]:
```suggestion
  Merge the following configuration into your custom values file:
```

File: docs/sources/release-notes/v2.4.md (hunk 2 above)
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md]:
```suggestion
- Updated the anti-affinity rules in the sizing plans (`small.yaml`, `large.yaml`, `capped-small.yaml`, `capped-large.yaml`) to account for the relaxed default rules.
```

File: docs/sources/release-notes/v2.4.md (hunk 3 above)
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md]:
```suggestion
- Added support for Openshift Route definition for nginx.
```

File: docs/sources/release-notes/v2.4.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+                  values:
+                    - store-gateway
+            topologyKey: "kubernetes.io/hostname"
+  ```
+- Updated the anti affinity rules in the sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml) to account for the relaxed default rules.
+- Support for Openshift Route definition for nginx has been added.
+
+## Important changes
+
+In Grafana Mimir 2.4 we have changed the default values of the following configuration options:
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:91]:
```suggestion
In Grafana Mimir 2.4, the default values of the following configuration options have changed:
```

File: docs/sources/release-notes/v2.4.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+- `-distributor.remote-timeout` has been changed from `20s` to `2s`.
+- `-distributor.forwarding.request-timeout` has been changed from `10s` to `2s`.
+- `-blocks-storage.tsdb.head-compaction-concurrency` has been changed from 5 to 1.
+- The hash ring heartbeat period for distributors, ingesters, rulers and compactors has been increased from `5s` to `15s`.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:96]:
```suggestion
- `-distributor.remote-timeout` has changed from `20s` to `2s`.
- `-distributor.forwarding.request-timeout` has changed from `10s` to `2s`.
- `-blocks-storage.tsdb.head-compaction-concurrency` has changed from `5` to `1`.
- The hash-ring heartbeat period for distributors, ingesters, rulers, and compactors has increased from `5s` to `15s`.
```

File: docs/sources/release-notes/v2.4.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+With Grafana Mimir 2.4 we have enabled the [anonymous usage statistics tracking]({{< relref "../operators-guide/configure/about-anonymous-usage-statistics-reporting.md" >}}) by default.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.4.md:98]:
```suggestion
With Grafana Mimir 2.4, the [anonymous usage statistics tracking]({{< relref "../operators-guide/configure/about-anonymous-usage-statistics-reporting.md" >}}) is enabled by default.
```

**Review by osg-grafana — APPROVED** [review]
Approving with feedback

**Comment by pracucci** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Comment by pracucci** [comment]
I need to publish 2.4.0-rc.0, so I will move on with this PR (since I haven't heard back from @pianohacker yet) but we can change / rewrite the release notes any time before the final 2.4.0 release.

File: docs/sources/release-notes/v2.4.md (hunk 9 above)
Comment (osg-grafana) [reply docs/sources/release-notes/v2.4.md:26]: Per our DMs, changing the suggestion now. :)

**Pull request merged**